### DISCUSSION TITLE
Cobwebs tricking Frigate into thinking my garden sofa is 82% human

### DISCUSSION THREAD

**Discussion by adamburgoyne** [blakeblackshear/frigate#11017] [General]
I've been using Frigate for well over a year without issues with my Reolink doorbell but I've just added a Reolink Duo 2 to the side of my house and I'm getting fairly frequent spider web problems (thanks to the IR presumably), these are causing interesting detections which is sadly limiting me from setting up night time alarm alerts when a person is detected whilst we're asleep.  

I woke up to about 30 alerts the other morning with the below snapshot. I can understand that the cobweb might be causing motion detections but I don't really understand what part of the image would be seen to be 82% person. Sometimes I can see that a shape/spider might _mildly_ resemble a person but in this case I have absolutely no idea, especially not with such high certainty. 

I think the answer is to probably go out every few days with the extending feather duster (I don't want to add an external IR blaster) but I just wanted to see if there's anything I can do config wise or is this perhaps something Frigate+ would/could address? I can't really ask friends to go round feather-dusting my cameras whilst on holiday is the main problem! 

![Screenshot_20240417_204430_Chrome](https://github.com/blakeblackshear/frigate/assets/1240218/4bd43ee1-2c2f-442d-b478-3e1ed51ce1e1)

**Comment by NickM-27** [comment]
Personally I saw this last summer and after submitting ~100 or so false positives to Frigate+ my model had greatly reduced false positives from these and I only got one every week or so which gave me enough time to clean anyway.

**Reply by adamburgoyne** [reply]
Sadly I'm not on Frigate+ but that's encouraging. I might have to (if possible) to join to get this sorted in that case, no idea why the cobweb itself causes the sofa to become a person, it's there all day without problem but as soon as the cobweb appears...

**Reply by NickM-27** [reply]
You can read more about it here https://github.com/blakeblackshear/frigate/discussions/7932

the cobweb is most likely just moving in the wind causing lots of motion which makes frigate look at the sofa many times eventually leading to it getting detected.

**Reply by adamburgoyne** [reply]
Interesting, I'm surprised even if it looked at it 1,000,000 times it would be identified as a person given it's shape doesn't remotely resemble one!

**Reply by NickM-27** [reply]
you have to think about the dataset that the model was trained on. the default model was trained on the COCO dataset which does not look like security cameras. [Here are some examples of images used to train the COCO models](https://imgur.com/a/CNImofJ)

So it is actually surprising the model works as well as it does

**Reply by adamburgoyne** [reply]
That's good to know and probably explains why then, perhaps it thinks the sofa cover is just a very large man! 

I've not been keeping track of the roadmap for Frigate but if the above dataset isn't going to change in the standard version I guess I'm going to have to look at subscribing to Frigate+ for a little while to get my own models at least. I'll read up a bit more on how it works to see if my understanding is correct. My doorbell however is extremely accurate with the default model, works very well with hardly any false positives - it is a much better view though given it just points at the driveway at face level!

**Reply by nwish** [reply]
Go out there at night with a few people and see what the average confidence and average area are for the “person” object, then adjust the threshold and max_area accordingly.  It looks like the camera is high enough to the point where someone standing as close to it as possible still wouldn’t fill more of the frame than the couch.  This is purely an assumption, and your mileage may vary, but it may help.
